proto/iotics/api/search.proto

```diff
@@ -83,6 +86,33 @@ message SearchRequest {
   Range range = 20;
 }
 
+// AdvancedSearchRequest describes a search request with more filtering possibilities than SearchRequest. It returns
+// formatted details about the twins matched by the supplied filter.
+message AdvancedSearchRequest {
+  // Search request payload.
+  message Payload {
+
+    // Expected response type
+    ResponseType responseType = 1;
+
+    // The search filter, expressed as a JSON-encoded AST (in JSONLogic style)
+    string filter = 2;
+  }
+
+  // Search request headers
+  Headers headers = 1;
+
+  // Search request scope
+  Scope scope = 2;
+
+  // Search request payload
+  Payload payload = 3;
+
+  // Search result limit (optional), affecting the maximum number of twins returned, per host. If not specified, a host-
+  // configured maximum limit will be applied.
+  Limit limit = 4;
```

Comment:
Offset is mising

Reply:
As discussed, I'll add offset (or rather: `Range`) in. If we choose to handle pagination/streaming differently in the future then both old & new (advanced) search will be affected in the same way (and we can deprecate/support them in similar fashion).